### PR TITLE
Cache chainID - set it on start

### LINKED ISSUE #405: Cache chainId instead of fetching it every time
Currently,
```go
var chain *big.Int
chain, e := ew.evmClient.ChainID(context.Background())
if e != nil {
  ew.logger.Errorf("[%s] - Failed to retrieve chain ID.", eventLog.Raw.TxHash)
  return
}
```
is called on every EVM related transfer, when this is data that can be cached upon validator start. It is not expected to change. Suggestion would be to cache it in`evmClient`.

### PR DESCRIPTION
**Pull request by svetlio8**
Signed-off-by: svetlio <svetlin.madzharov@gmail.com>

**Detailed description**:
Cache chainID - set it on start

**Which issue(s) this PR fixes**:
Fixes #405

**Checklist**
- [x] Tests updated

### CODE REVIEW

File: cmd/clients.go [hunk 1]
```diff
@@ -34,8 +36,19 @@ type Clients struct {
...
+		if e != nil {
+			log.Fatalf("[%d] - Failed to retrieve chain ID on client prepare.", configChainId)
+			return nil
+		}
+		if configChainId == clientChainId.Uint64() {
+			EVMClients[configChainId].SetChainID(clientChainId.Uint64())
+		} else {
+			log.Fatalf("Chain IDs mismatch [%d] config, [%d] actual.", configChainId, clientChainId)
+			return nil
+		}
```
Comment (failfmi) [inline cmd/clients.go:51]:
```go
		if configChainId != clientChainId.Uint64() {
			log.Fatalf("Chain IDs mismatch [%d] config, [%d] actual.", configChainId, clientChainId)
		}
		EVMClients[configChainId].SetChainID(clientChainId.Uint64())
```

Comment (failfmi) [inline cmd/clients.go:51]:
```go
		if configChainId != clientChainId.Uint64() {
			log.Fatalf("Chain IDs mismatch [%d] config, [%d] actual.", configChainId, clientChainId)
		}
		EVMClients[configChainId].SetChainID(clientChainId.Uint64())
```

File: cmd/clients.go [hunk 2]
```diff
@@ -34,8 +36,19 @@ type Clients struct {
...
 func PrepareClients(config config.Clients) *Clients {
 	EVMClients := make(map[uint64]client.EVM)
-	for chainId, ec := range config.Evm {
-		EVMClients[chainId] = evm.NewClient(ec)
+	for configChainId, ec := range config.Evm {
+		EVMClients[configChainId] = evm.NewClient(ec, configChainId)
+		clientChainId, e := EVMClients[configChainId].ChainID(context.Background())
+		if e != nil {
+			log.Fatalf("[%d] - Failed to retrieve chain ID on client prepare.", configChainId)
+			return nil
```
Comment (failfmi) [inline cmd/clients.go:44]: l`og.Fatalf` exits the program There is no need to return anything.

Comment (svetlio8) [reply cmd/clients.go]: done

**Review by failfmi — APPROVED** [review]

**Pull request merged**